### PR TITLE
chore: add missing lifecycles interfaces to declarations

### PR DESCRIPTION
**Pull request by rafaelss95**
Some declarations weren't implementing the `LifeCycle`s hooks as `OnDestroy`, `OnInit`, etc. It's not necessary, much less a problem, but since in most declarations you usually use it, it's good to maintain the consistency.

Note that I added [codelyzer](https://github.com/mgechev/codelyzer/issues), which prevents these mistakes, let me know if you guys agree to add it or if I should remove it from this PR.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -85,6 +86,14 @@
     "jsdoc-format": [true, "check-multiline-start"],
     "no-duplicate-imports": true,
 
+    // Codelyzer
+    "banana-in-box": true,
+    // TODO: enable when https://github.com/mgechev/codelyzer/issues/545 is solved
+    "contextual-life-cycle": false,
+    "decorator-not-allowed": true,
+    "templates-no-negated-async": true,
+    "use-life-cycle-interface": true,
```
Comment (jelbourn) [inline tslint.json]: Can you explain what these do? I'm not familiar with the specific checks.

File: tslint.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "rulesDirectory": [
-    "./tools/tslint-rules/"
+    "./tools/tslint-rules/",
+    "node_modules/codelyzer"
```
Comment (jelbourn) [inline tslint.json]: Will this enforce anything not explicitly listed below?

File: tslint.json (hunk 1 above)
Comment (rafaelss95) [reply tslint.json]: Of course.

**banana-in-box**: Ensure that the two-way data binding syntax is correct.
**contextual-life-cycle**: Ensure that classes use allowed life cycle method in its body, ex.:
Injectable classes should not implement or declare `ngOnInit` hook.
**decorator-not-allowed**: Ensure that classes use allowed decorator in its body. ex.:
Injectable classes should not use `@Input`.
**no-output-on-prefix**: Disallow outputs with the prefix "on".
**templates-no-negated-async**: Ensures that strict equality is used when evaluating negations on async pipe output.
**use-life-cycle-interface**: Ensure that components implement life cycle interfaces if they use them. (This is what caught the failures that are being fixed in this PR).

File: tslint.json (hunk 2 above)
Comment (rafaelss95) [reply tslint.json]: Sorry for the delayed response. No, it won't enforce anything unless you explicit say what you want.

**Comment by rafaelss95** [comment]
@jelbourn Rebased.

**Comment by rafaelss95** [comment]
@jelbourn could you please remove the `needs rebase`?

**Pull request merged**